### PR TITLE
Fix hardcoded node names in testrunner

### PR DESCRIPTION
**Pull request by pablochacin**
## Why is this PR needed?

When the cloud-provider option is used for bootstrapping a cluster, the process fails because testrunner invokes skuba bootstrap command with a node name that doesn't match the name expected by openstack. 

Fixes https://github.com/SUSE/avant-garde/issues/1000

## What does this PR do?

The hardcoded names used in testrunner code are changes by names provided by the Platform used to provision the cluster. In the case of openstack, the name are according to the expected name by the cloud provider.

This change also impacted some tests that used hardcoded node names.  The tests were fixed and simplified removing duplicated code that was affected by the changes. 

Finally, the logic used for checking the nodes are ready after update was also change to make tests run more reliably.
 
## Anything else a reviewer needs to know?

Some changes in the tests were not strictly required for the purpose of this PR but required to make tests run reliably.

Some tests are failing, but for reasons that seems not to be related to this PR. Those issues will be addressed in follow up PRs.

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/infra/testrunner/skuba/skuba.py [hunk 1]
```diff
@@ -105,9 +107,11 @@ def join_nodes(self, masters=None, workers=None):
 
         for n in range(1, masters):
             self.node_join("master", n)
+            time.sleep(60*3)
```
[the remark this answers is not in the document]
Comment (mjura) [reply ci/infra/testrunner/skuba/skuba.py]: I think we could provide this as argument

File: ci/infra/testrunner/skuba/skuba.py [hunk 2]
```diff
@@ -105,9 +107,11 @@ def join_nodes(self, masters=None, workers=None):
...
 
         for n in range(0, workers):
             self.node_join("worker", n)
+            time.sleep(60*3)
```
[the remark this answers is not in the document]
Comment (mjura) [reply ci/infra/testrunner/skuba/skuba.py]: I think we could provide this as argument

File: ci/infra/libvirt/terraform.tfvars.json.ci.example [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "libvirt_uri": "qemu:///system",
     "pool": "default",
-    "image_uri": "SLES15-SP1-JeOS.x86_64-15.1-OpenStack-Cloud-GMC3.qcow2",
+    "image_uri": "/home/pchacin/Downloads/SLE-15-SP1-JeOS-GMC.qcow2",
```
[the remark this answers is not in the document]
Comment (mjura) [reply ci/infra/libvirt/terraform.tfvars.json.ci.example]:
```suggestion
    "image_uri": "SLE-15-SP1-JeOS-GMC.qcow2",
```

File: ci/infra/testrunner/skuba/skuba.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pablochacin) [reply ci/infra/testrunner/skuba/skuba.py]: This is a hack and shouldn't be in this PR (there's another PR that addresses this issue). Removing it.

File: ci/infra/testrunner/skuba/skuba.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pablochacin) [reply ci/infra/testrunner/skuba/skuba.py]: Same here

**Pull request merged**